backend/src/routes/chat/chat.gateway.ts

```diff
@@ -67,39 +68,34 @@ export class ChatGateway implements OnGatewayInit, OnGatewayConnection {
     @ConnectedSocket() client: Socket,
     @MessageBody("channel") channelName: string
   ) {
+    if (!client || !channelName) return;
     const channel = client.request.channels.find((x) => x.name == channelName);
     if (!channel) return;
 
     const username = client.request.user.displayname;
-
+    client.join(client.request.user.id.toString());
     client.join(channel.id);
+    console.log(client.request.user.id.toString());
     client.to(channel.id).emit("newMessage", {
       message: `${username} joined the room`,
       login: "ADMIN",
       displayname: "ADMIN",
       id: 0,
     });
   }
-  @SubscribeMessage("sendpm")
-  async handlepm(
-    @ConnectedSocket() client: Socket,
-    @MessageBody("name") receiverName: string,
-    @MessageBody("str") senderlogin: string
-  ) {
-    if (receiverName === senderlogin) return;
-    const receiver = await this.userService.getUserByLogin(senderlogin);
-    if (!receiver) return;
-    client.to(String(receiver?.id)).emit("pm", { test: String, test2: String });
-  }
   @SubscribeMessage("sendMessage")
   async handleEvent(
     @ConnectedSocket() client: Socket,
     @MessageBody("channel") channelName: string,
     @MessageBody("msg") message: string
   ) {
+    if (!client || !channelName || !message) return;
     const channel = client.request.channels.find((x) => x.name == channelName);
     if (!channel) return;
-
+    if (await this.channelBanService.isBanned(client.request.user.id)) {
```

Comment:
We probably shouldn't check for bans here, and instead direcly kick people of the channel when they're banned.